tests/input_converter/test_converter.py

```diff
@@ -618,3 +720,246 @@ def test_convert_links_to_component(self, local_study_w_links):
 
         assert links_components == expected_link_component
         assert links_connections == expected_link_connections
+
+    def _generate_tdp_instance_parameter(
```

Comment:
This function does too many things, split th creation of timeseries from the part that gets and returns the tdp object